content/blog/dplyr-1-1-0-joins/index.Rmd

```diff
@@ -0,0 +1,237 @@
+---
+output: hugodown::hugo_document
+slug: dplyr-1-1-0-joins
+title: "dplyr 1.1.0: Joins"
+date: 2023-01-31
+author: Davis Vaughan
+description: >
+    In dplyr 1.1.0, joins have been greatly reworked, introducing: a new way to
+    specify join columns, various new types of joins, and two new quality
+    control arguments.
+photo:
+  url: https://unsplash.com/photos/Cecb0_8Hx-o
+  author: Duy Pham
+categories: [package] 
+tags: [dplyr]
+editor_options: 
+  chunk_output_type: console
+---
+
+[dplyr 1.1.0](https://dplyr.tidyverse.org/news/index.html#dplyr-110) is out now!
+This is post 1 of 4 detailing some of the new features in this release.
+In this post, we will discuss various new updates to dplyr's joins.
+
+You can install it from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("dplyr")
+```
+
+```{r setup, warning=FALSE, message=FALSE}
+library(dplyr)
+```
+
+## `join_by()`
+
+Consider the following two tables, `transactions` and `companies`.
+`transactions` tracks sales across various years for different companies, and `companies` connects the short company id to its actual company name - either Patagonia (a fellow B-Corp!) or RStudio.
+
+```{r}
+transactions <- tibble(
+  company = c("A", "A", "B", "B"),
+  year = c(2019, 2020, 2021, 2023),
+  revenue = c(50, 4, 10, 12)
+)
+transactions
+
+companies <- tibble(
+  id = c("A", "B"),
+  name = c("Patagonia", "RStudio")
+)
+companies
+```
+
+To join these two tables together, we might use an inner join:
+
+```{r}
+transactions |> 
+  inner_join(companies, by = c(company = "id"))
+```
+
+This works great, but has always felt a little clunky.
+Specifying `c(company = "id")` is a little unnatural for new users, especially if they are used to "equivalence" in R being expressed with `==`.
```

Comment:
```suggestion
Specifying `c(company = "id")` is a little awkward because it uses `=`, not `==`: here we're asserting that we want `company` to equal `id`, not naming a function argument or doing assignment.
```